### ISSUE TITLE
Grafana 4.7 datasource from file issue

### ISSUE THREAD

**Issue by calebcall** [grafana/grafana#10200]
- What did you do? Upgraded to grafana-4.7.0-10747pre1.x86_64 from 4.6.2
- What happened instead? It wouldn't start with the following error: lvl=eror msg="Startup failed" error="Failed to provision Grafana from config. error: open /etc/grafana/provisioning/datasources: no such file or directory"

I then created the /etc/grafana/provisioning directory (wasn't there previously), uninstalled/reinstalled and it then worked.

**Comment by calebcall** [comment]
Little more info on this, I upgraded a second server and preemptively created the /etc/grafana/provisioning directory.  Upgrade still failed with the same message as above.  I then did nothing more than remove the rpm and then reinstalled the rpm and it then created the datasources and dashboards directories.

**Comment by bergquist** [comment]
@calebcall did Grafana have read access to that folder?

**Comment by calebcall** [comment]
The first install, it had full r/w access to /etc/grafana (as mentioned, the provisioning directory didn’t exist. On the second one that I manually created the provisioning directory I also changed the owner to grafana and made sure it had r/w access.

**Comment by bergquist** [comment]
Grafana now handles missing provision folders more gracefully so this is not an issue anymore.

**Comment by Vehyla** [comment]
Using grafana 5.0 I am getting the same error.  Perhaps it is still an issue?

**Comment by bergquist** [comment]
@Vehyla please share more info about dist and how your running Grafana.

**Comment by Vehyla** [comment]
We are using an ec2 Amazon linux 2017.09.01 AMI.  I just did a simple rpm install, setup my ldap from a known config then changed out the host-name info.  I realized later that my real error was the wrong redis port.  Which caused the daemon to stop and one of the last messages was:

t=2018-03-07T17:10:13+0000 lvl=eror msg="cant read datasource provisioning files from directory" logger=provisioning.datasources path=/etc/grafana/provisioning/datasources
t=2018-03-07T17:10:13+0000 lvl=eror msg="cant read dashboard provisioning files from directory" logger=provisioning.dashboard path=/etc/grafana/provisioning/dashboards

I ran the services manually saw redis errors than configured the correct port and grafana came up.  But even if I do a restart now on with the working config this error pops up in the logs but does not create the directory.  So now I see the error.  But it's not causing any issues, so I would say it was not something that actually causes failures or is critical to running the product.
